### ISSUE TITLE
work on ICE demo application

### ISSUE THREAD

**Issue by jekram** [Cloudkibo/ice#5]
Do a demo on Ice. Please post status and issues here

**Comment by sojharo** [comment]
Up till now, the current website looks like this. It is not able to get the key MilliSecs from file.
I am going to post this as question on their repository.

![ic3-3](https://cloud.githubusercontent.com/assets/5811465/7767428/e980adf2-008b-11e5-9246-bb4a5ae3fa4a.PNG)
![ice-1](https://cloud.githubusercontent.com/assets/5811465/7767430/e983e2d8-008b-11e5-8500-ad27b2b8c3e6.PNG)
![ice-2](https://cloud.githubusercontent.com/assets/5811465/7767429/e9816a9e-008b-11e5-9fa7-5278238b90cd.PNG)

**Comment by jekram** [comment]
It lol like this is a dats issue and not logic. We have missing dats

**Comment by sojharo** [comment]
I worked on it and it does not give any apparent error now. Both errors got resolved by working on ice.properties file. Still It does not show data.

**Comment by jekram** [comment]
Please do a logging trace to figure it out

**Comment by sojharo** [comment]
I have looked more into it. As earlier it does not give any apparent error as long as code to do reservation polling is commented out. We had commented out this code earlier as it gave the error and wanted the authorization rights on Amazon EC2 services.

It looks like that Amazon EC2 service code should not be commented out. I have uncommented it and here is the actual output:

| Error 2015-05-29 16:26:28,764 [localhost-startStop-1] ERROR basic.BasicReservationService  - failed to poll reservation prices
	Message: You are not authorized to perform this operation. (Service: AmazonEC2; Status Code: 403; Error Code: UnauthorizedOperation; Request ID: e3b8372d-16a5-423b-9991-d86152961cb3)
		Line | Method
	->> 1077 | handleErrorResponse                in com.amazonaws.http.AmazonHttpClient
	- - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - - -
	|    725 | executeOneRequest                  in     ''
	|    460 | executeHelper . . . . . . . . . .  in     ''
	|    295 | execute                            in     ''
	|   9393 | invoke . . . . . . . . . . . . . . in com.amazonaws.services.ec2.AmazonEC2Client
	|   7020 | describeReservedInstancesOfferings in     ''
	|    152 | pollAPI . . . . . . . . . . . . .  in com.netflix.ice.basic.BasicReservationService
	|    106 | init                               in     ''
	|     95 | <init> . . . . . . . . . . . . . . in com.netflix.ice.processor.ProcessorConfig
	|    186 | doCall                             in BootStrap$_closure1
	|    327 | evaluateEnvironmentSpecificBlock . in grails.util.Environment
	|    320 | executeForEnvironment              in     ''
	|    296 | executeForCurrentEnvironment . . . in     ''
	|    266 | run                                in java.util.concurrent.FutureTask
	|   1142 | runWorker . . . . . . . . . . . .  in java.util.concurrent.ThreadPoolExecutor
	|    617 | run                                in java.util.concurrent.ThreadPoolExecutor$Worker
	^    745 | run . . . . . . . . . . . . . . .  in java.lang.Thread

We are just trying to do the demo so let us give the authorization and rights to billing account for Amazon EC2 service. Let's do it with all the access rights which ice requires for now. Later, we would remove the rights from code that we don't need.

Till we do this, I am going to fix the issue on ice droplet to run it on live server. There is some strange java error that I am trying to fix.

**Comment by sojharo** [comment]
Also, I found google group of ice project, on the error we have, they also say that we need to give the permission:

https://groups.google.com/forum/#!topic/iceusers/0hVrM-balMc

**Comment by jekram** [comment]
I will later look more in permission. 

Thanks for looking into thid

**Comment by jekram** [comment]
Asak

Please retry and respond. You have full access to S3 and ec2

Jawsid
________________________________

**Comment by jekram** [comment]
Sojharo asak:

You have full access to EC2 so please retry.

It looks like even though you are running on a local host it is trying to provision an vm on a dynamic basis. So they much have some place hardcoded to create an VM instance in AWS and it is failing since you did not had the access right to EC2 of AWS.

**Comment by sojharo** [comment]
It does not give the EC2 authorization error and was able to proceed. It crashed and said:

Failed to write core dump. Minidumps are not enabled by default on client versions of Windows

I am trying to enable Minidumps on Windows. Also I am trying to do this on live server which is linux.

**Comment by sojharo** [comment]
I have found the root cause of the error. It is error in JRE itself and it happens only on Windows:

https://java.com/en/download/help/exception_access.xml

I have tried their given solution. It did not work. I am now trying on our droplet which runs the Linux.

**Comment by sojharo** [comment]
I started the ice server on our droplet, it did a long processing and made the following files in the bucket.

![bucketchanged](https://cloud.githubusercontent.com/assets/5811465/7896908/66828eb8-06e2-11e5-87a3-75f6861b3eec.PNG)

It is still processing, once it is done, i would load the website and put snapshots here. Looks like, we have a progress on this.

**Comment by sojharo** [comment]
On the live server, it runs successfully without any errors but still it does not show any data or charts. There is no apparent error on client side too:

![liveserverice](https://cloud.githubusercontent.com/assets/5811465/7897045/e9c2e792-06e6-11e5-84f8-824e0eed99da.PNG)

**Comment by sojharo** [comment]
I stopped the reader and just ran the processor and after some work and changes in ice.properties file, I finally managed to see the required files uploaded to our s3 bucket. These files are created by the processor and now I would run the reader and reader would use these files.

![bucketnewfiles](https://cloud.githubusercontent.com/assets/5811465/7897516/d3b76746-06f7-11e5-9654-74ce1a1638d4.PNG)

**Comment by sojharo** [comment]
I finally managed to get the data through reader on the UI. It shows the value $0 for all the filters. It does not make the chart. Maybe, because of 0 value it does not show chart. It shows the aggregated values too.

![success](https://cloud.githubusercontent.com/assets/5811465/7897670/50e17036-06fd-11e5-822c-255b269db55b.PNG)

![success2](https://cloud.githubusercontent.com/assets/5811465/7897671/5a92fbe0-06fd-11e5-9060-3640efe28917.PNG)

Now, next thing is to figure out how I can run the live server forever, even if I close the droplet from my side it should be running there. Currently, when I close the droplet, it closes this server  too. This used to happen with nodejs too then we figured out how we can make it live forever.

http://45.55.232.65:8080/ice/dashboard/summary

Use the above address to see the application working on live server. I would keep the computer open so that live server is not down untill I figure out the way to run in production mode so that it runs forever.

**Comment by jekram** [comment]
Fantastic job. :-)

**Comment by jekram** [comment]
ASAK:

I could not see the ICE server at the address above. Let's just do a demo when we meet next.

The reason it is showing zero because there is no charges in the current system.

This is very good progress.

Next steps:

1. Can we upload a csv file with fake records. Send me the current file and I will create a billing file.
2. Close ice to iceazure (keep it in public for now). Modify the logic to process Azure billing. For Azure billing let's load our own csv file. They will be loaded into a separate droplet.
3. Do the same for icegoogle. 
4. A web page that will toggle between the three droplets
5. Maybe a summary page with to show 12 months of charges for each cloud.

Allah Hafiz

Jawaid

**Comment by sojharo** [comment]
Yes, the server gets closed when I close the droplet access from here. I am trying to find a way that it keeps running in background as a service and doesn't close when I leave the droplet.

**Comment by sojharo** [comment]
We need to use ICE cookbook in order to deploy the application to the server. Currently, I was running development mode of the ice application on our server which was closed when I closed the server from my side. Here is the link for ICE cookbook:

https://github.com/mdsol/ice_cookbook

It is recommended that we run ice reader and ice processor on different servers. However, it also works if we run both in one server with right configurations. I managed to do that right configuration and ran both processor and reader on same server.

I am going to look into ice cook book in order to understand how we can deploy ice application. For demo purpose, I would run the development mode of application today on live server.

**Comment by jekram** [comment]
Is the cookbook Chef cookbook?
________________________________

**Comment by sojharo** [comment]
Yes, it is a chef cookbook. I am trying to understand how chef works from this link:

https://supermarket.chef.io/cookbooks-directory

http://docs.chef.io/cookbooks.html

**Comment by sojharo** [comment]
Check this link now:

http://45.55.232.65:8080/ice/dashboard/summary

I have kept the server running on the droplet and droplet has been opened here. It would close as I close the server.

I could not understand how chef works, I need to look more into it today. Usually, in Groovy Grails, we build the application make *.war file out of it. This war file then is given to TomCat server and TomCat runs this file. I tried generating the war file and it gave some version error.

For ice, it seems little different and we need to do it using chef.
